packages/amplication-client/src/Modules/hooks/useModule.tsx

```diff
@@ -0,0 +1,146 @@
+import { Reference, useLazyQuery, useMutation } from "@apollo/client";
+import { useContext } from "react";
+import { AppContext } from "../../context/appContext";
+import * as models from "../../models";
+import {
+  CREATE_MODULE,
+  DELETE_MODULE,
+  FIND_MODULES,
+  GET_MODULE,
+  MODULE_FIELDS_FRAGMENT,
+  UPDATE_MODULE,
+} from "../queries/modulesQueries";
+type TDeleteData = {
+  deleteModule: models.Module;
+};
+
+type TFindData = {
+  Modules: models.Module[];
+};
+
+type TGetData = {
+  Module: models.Module;
+};
+
+type TCreateData = {
+  createModule: models.Module;
+};
+
+type TUpdateData = {
+  updateModule: models.Module;
+};
+
+const useModule = () => {
+  const { addBlock } = useContext(AppContext);
+
+  const [
+    deleteModule,
+    { error: deleteModuleError, loading: deleteModuleLoading },
+  ] = useMutation<TDeleteData>(DELETE_MODULE, {
+    update(cache, { data }) {
+      if (!data || data === undefined) return;
+      const deletedModuleId = data.deleteModule.id;
+      cache.modify({
```

Comment:
again using apollo cache instead of react context

Reply:
@yuval-hazaz   FYi